demo/core_34_$.www.code.js

```diff
@@ -145,28 +145,35 @@ $.www.code.editor.www = function(request, response) {
   data.saved = false;
   var parts = JSON.parse(request.parameters.parts);
   if (parts.length) {
-    // Find the origin object.  '$.foo' is the origin of '$.foo.bar'.
     var lastPart = parts.pop();
     var object;
-    if (request.parameters.key) {
-      // See if temp ID DB still knows about this key.  If so, we're done.
-      object = $.db.tempId.getObjById(request.parameters.key);
-    }
-    if (!object) {
-      // Parse the parts list.
-      try {
-        var object = $.utils.selector.partsToValue(parts);
-      } catch (e) {
-        // Parts don't match a valid path.
-        data.butter = 'Unknown object';
+    var isGlobal = parts.length < 1;
+    if (isGlobal) {
+      object = $.utils.selector.getGlobal();
+    } else {
+      // Find the origin object.  '$.foo' is the origin of '$.foo.bar'.
+      if (request.parameters.key) {
+        // See if temp ID DB still knows about this key.  If so, we're done.
+        object = $.db.tempId.getObjById(request.parameters.key);
+      }
+      if (!object) {
+        // Parse the parts list.
+        try {
```

Comment:
This section is all fairly arbitrarily nested.  It might be easier to read if it were structured as follows:
```JS
if (!object && condition1) {
  object = whatever1;
}
if (!object && condition2) {
  object = whatever2;
}
...
```